src/__tests__/fightClass.spec.ts

```diff
@@ -1,14 +1,16 @@
 import { Fight } from "../fightClass"
-
+import { PokemonType } from "../pokemonClass"
+import { PokemonFactory } from "../pokemonFactory"
+import * as data from "../../assets/poke_data.json"
 
 const fight = new Fight();
 
 describe('Test fight class', () => {
 
     it("Should return effectivenes of the attack for a given attack type and pokemon defending type", () => {
         // Given
-        const attackTypes: string[] = ["normal", "electric", "fire"]
-        const pokemonTypes: string[][] = [["normal"], ["normal", "water"], ["bug", "grass"]]
+        const attackTypes: PokemonType[] = ["normal", "electric", "fire"]
+        const pokemonTypes: PokemonType[][] = [["normal"], ["normal", "water"], ["bug", "grass"]]
```

Comment:
Why not separate this into three different tests?
I see there three tests:
"When there is no change"
"When there is x2 damage"
"When there is x4 damage"

Reply:
The other way is to create **parameterised tests** here :)
See:
https://medium.com/@suman.maity112/parameterized-tests-with-jest-2932f42b1509

Reply:
The idea is to start with the most simple test: "Two pokemons, one attack another, no modifires/effectiveness" then add more complicated ones: "fire vs water will be 0,5 times damage", "water vs fire will be 2 times damage", "water vs [fire, rock] will be X times damage" etc.